packages/vite/src/node/server/index.ts

```diff
@@ -821,10 +822,12 @@ async function restartServer(server: ViteDevServer) {
   } = server.config
   if (!middlewareMode) {
     await server.listen(port, true)
+    const dnsOrderChange = diffDnsOrderChange(oldUrls, newServer.resolvedUrls)
     logger.info('server restarted.', { timestamp: true })
     if (
       (port ?? DEFAULT_DEV_PORT) !== (prevPort ?? DEFAULT_DEV_PORT) ||
-      host !== prevHost
+      host !== prevHost ||
+      dnsOrderChange
```

Comment:
It is better to move the function call to the condition so we allow short-circuiting it

Reply:
Done.